app/services/marketplace_service/store/search_settings.rb

```diff
@@ -0,0 +1,49 @@
+module MarketplaceService::Store::SearchSettings
+
+  SearchSettingsModel = ::SearchSettings
+
+  SearchSettings = EntityUtils.define_builder(
+    [:community_id, :mandatory, :fixnum],
+    [:main_search,:to_symbol, :mandatory, one_of: [:keyword, :location]]
```

Comment:
`:mandatory` is not mandatory (pun intended). `one_of` guarantees that the value is not `nil`